upload.go

```diff
@@ -34,15 +34,22 @@ func Upload(file *os.File) (string, error) {
 	}
 
 	cfg := aws.NewConfig().WithRegion(awsRegion).WithCredentials(creds)
-	svc := s3.New(session.New(), cfg)
+	sesh, err := session.NewSession()
+	if err != nil {
+		log.Fatalf("failed to create AWS session: %v", err)
+	}
+	svc := s3.New(sesh, cfg)
 
 	fileInfo, err := file.Stat()
 	if err != nil {
 		log.Fatalf("%v", err)
 	}
 
 	buffer := new(bytes.Buffer)
-	file.Seek(0, 0)
+	_, err = file.Seek(0, 0)
```

Comment:
Excellent find to check the `err`. Is there a better way than using `Seek` here?